src/modules/site/base/utils/auth.py

```diff
@@ -81,9 +97,22 @@ def unauthorized_callback(reason):
   return redirect(url_for('auth.choose_login')), 302
 
 
+def check_if_token_revoked(jwt_data):
+    jti = jwt_data["jti"]
+    user_token = UserToken(jti)
+    token_revoked = user_token.revoked
+    if token_revoked:
+      logger.info("Token has been revoked.")
+      return make_response(redirect(url_for('auth.choose_login'))), 302
+    logger.info(f"Revoking user token - {jti} for user - {user_token.username}")
+    user_token.revoke()
+    return token_revoked
+    
+
 @jwt.invalid_token_loader
 def invalid_token_callback(callback):
   ''' Invalid Fresh/Non-Fresh Access token in auth header, redirect to login '''
+  logger.info("Invalid Token.")
   resp = make_response(redirect(url_for('auth.choose_login')))
```

Comment:
consider adding a 'flash'/banner explaining what happened to the token to the user.